### DISCUSSION TITLE
Blog #2

### DISCUSSION THREAD

**Discussion by AlinaBoess** [AlinaBoess/SoftwareEngineeringProjektTINF23B5#2] [General]
# Zweiter Blogeintrag

Schön, dass ihr es wieder zu uns geschafft habt :)
Hier sind ein paar Updates zu unserem Projekt:

## Software Requirement Specification (SRS)
Wir haben unsere SRS geschrieben und dokumentiert. 
Ihr könnt euch dies unter [SRS](https://github.com/AlinaBoess/SoftwareEngineeringProjektTINF23B5/blob/main/documentation/SoftwareRequirementDocumentation.md) anschauen.
Bitte beachtet hierbei, dass wir mit unseren Use-Cases nch nicht ganz fertig sind. Wir werden diese Stückweise ergänzen!
Auch unsere Mockups werden wir Stück für Stück noch ergänzen, da wir uns leider noch nicht ganz auf ein Design einigen konnten.

## sonstige Aufgaben
Ansonsten haben wir uns noch hiermit beschäftigt:
- Strukturierung unseres GitHub Repositorys (Aufteilung in Doku, Code und unfertige Blogs)
- Installation der Entwicklungsumgebung (Visual Studio 2022 Community)
- Familiarisierung mit Git und dessen Umgang
- Weiterführende Organisation des Projektes in unserem Jira

Schaut gerne beim nächsten Mal wieder vorbei, wir freuen uns auf euch ;)
Bei Wünschen oder Anregungen dürft ihr uns gerne schreiben, wir freuen uns auf euer Feedback!

Liebe Grüße 
Lukas, Alex, Moumen, Yahya und Alina

**Comment by SchloesserJonas** [comment]
Hallo zusammen,

das sieht ja schon mal echt gut aus!
Wie plant ihr eine Echtzeitdarstellung der freien/belegten Tische zu realisieren?
Wir hatten bereits bei einer weiteren Gruppe angemerkt das ein ständiges Abfragen (Polling) des Backends bzw. der Datenbank wenig effizient und eher schlechte Praxis ist.
Falls ihr diese Funktionalität beibehalten möchtet, macht euch mal mit Websockets vertraut. Diese bieten die Möglichkeit das der Server an den Client Daten sendet ohne das vorher ein Request des Clients erfolgt ist.
Alternativ programmiert euer Frontend so, dass nach 10sec (zum Beispiel) die Darstellung von einem Popup überdeckt wird, welches den Nutzer bittet die Seite neu zu laden, um die Ansicht zu aktualisieren.

Viel Erfolg weiterhin!

GaLiGrü 😊
Paula, Safae, Samuel, Valentin, Jonas

> Green Sprout 🌱

**Comment by Mimikry2** [comment]
Moin,

eure Software Requirement Documentation ist gut strukturiert und deckt die wichtigsten Aspekte ab! Besonders die klare Darstellung der Use Cases und die Beschreibung der Anforderungen sind hilfreich für das Verständnis des Projekts. Es wäre jedoch interessant, mehr über die Priorisierung der Anforderungen und eure Überlegungen zur User Experience zu erfahren. Wir sind gespannt, wie sich euer Projekt weiterentwickelt!

Viele Grüße,
Tim, Dennis, Nico

**Comment by GermanJesus-lul** [comment]
Hey,

eure Software Requirements Specification sieht echt gut aus.
Ich finde jeden Punkt gut formuliert und ausführlich beschrieben.
Aufgefallen ist mir aber, dass ihr das Auswählen/Suchen eines Restaurants in eurem Dokument nicht erwähnt/beschrieben habt.
Wie habt ihr geplant, dass ein Benutzer die Reservierungsseite für ein bestimmtes Restaurant findet? Ist das Aufgabe des Restaurants? Bspw. durch einen Link im Google Maps Eintrag.

Mit freundlichen Grüßen,
Das Rezeptbuch Team
